src/Tizen.NUI/src/public/Utility/GraphicsTypeManager.cs

```diff
@@ -28,13 +29,118 @@ public class GraphicsTypeManager
+        /// Constant of triple extra high(640) density dpi.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public const ushort DensityXXXHigh = 640;
+
+        /// <summary>
+        /// Custom scale factor of display metrics.
+        /// ScalingFactor scale Dpi on DpiStable.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public float ScalingFactor
+        {
+            get => scalingFactor;
+            internal set => scalingFactor = value;
+        }
+
+        /// <summary>
+        /// Dot per Inch value from system.
+        /// See Vector Dpi in <see cref="Tizen.NUI.Window" /> also.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int Dpi
+        {
+            get
+            {
+               Vector2 screenDpi = NUIApplication.GetDefaultWindow().Dpi;
+
+               // Currently Dpi.X and Dpi.Y is all same value from ecore_wl2_output_dpi_get
+               // Also Diagonal Dpi should be same as X, Y Dpi in normal rectangle-pixels display
+               // so for the convenience, we use Dpi.X
+               return Convert.ToInt32(Math.Round(screenDpi.X));
+            }
+        }
+
+        /// <summary>
+        /// Dpi for GraphicsTypeManager.
+        /// Dpi is scaled from Dpi with custom ScalingFactor.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int ScaledDpi
+        {
+            get
+            {
+                return Convert.ToInt32(Math.Round(Dpi * ScalingFactor));
+            }
+        }
+
+        /// <summary>
+        /// Default baseline dpi. Medium(160) density dpi is origianlly provided.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int BaselineDpi
+        {
+            get => baselineDpi;
+            internal set
+            {
+                baselineDpi = value;
+            }
+        }
+
+        /// <summary>
+        /// Density of display.
+        /// </summary>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public float Density
         {
-            typeConverter = new GraphicsTypeConverter();
+            get => ((float)BaselineDpi / (float)ScaledDpi);
```

Comment:
Density의 정의가 Android와 반대로 되어 있습니다.

https://developer.android.com/reference/android/util/DisplayMetrics#density

>The logical density of the display. This is a scaling factor for the Density Independent Pixel unit, where one DIP is one pixel on an approximately 160 dpi screen (for example a 240x320, 1.5"x2" screen), providing the baseline of the system's display. Thus on a 160dpi screen this density value will be 1; on a 120 dpi screen it would be .75; etc. 
```suggestion
            get => ((float)ScaledDpi / (float)BaselineDpi);
```

Reply:
밀도니까 말씀하신 정의가 맞네요. 감사합니다